src/ConnectedMode/Promote/PromoteGoldBar.cs

```diff
@@ -0,0 +1,100 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.ConnectedMode.UI;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Core.Notifications;
+using SonarLint.VisualStudio.Core.Telemetry;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Promote;
+
+public interface IPromoteGoldBar
+{
+    void PromoteConnectedMode(string languagesToPromote);
+}
+
+[Export(typeof(IPromoteGoldBar))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+public sealed class PromoteGoldBar : IPromoteGoldBar, IDisposable
+{
+    private readonly INotificationService notificationService;
+    private readonly IDoNotShowAgainNotificationAction doNotShowAgainNotificationAction;
+    private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+    private readonly IBrowserService browserService;
+    private readonly ITelemetryManager telemetryManager;
+    private readonly IConnectedModeUIManager connectedModeUiManager;
+
+    [ImportingConstructor]
+    public PromoteGoldBar(
```

Comment:
Elsewhere, except in one other place, we used the term `Notification` instead of `GoldBar`